src/app.ts

```diff
@@ -206,6 +186,36 @@ export default class ChessGame {
 		await Promise.all(preloads);
 	}
 
+	private resetGame() {
+		this.game = chess.createSimple();
+		this.game.on('check', (attack: Attack) => this.onCheck(attack));
+		this.loadActorsAndEvents();
+	}
+	
+	private async loadActorsAndEvents() {
+		if (this.sceneRoot) {
+			//destroy scene root to avoid duplicating assets
+			this.sceneRoot.destroy();
+		}
+		await this.preloadAllModels();
+		
+		// Create all the actors.
+		await Promise.all([
+			this.createRootObject(),
+			this.createChessboard(),
+			this.createChessPieces(),
+			this.createMoveMarkers(),
+			this.createCheckMarker(),
+			this.createJoinButtons(),
+			this.createResetButton()
+		]);
+
+		// Hook up event handlers.
+		// Do this after all actors are loaded because the event handlers themselves reference other actors in the
+		// scene. It simplifies handler code if we can assume that the actors are loaded.
+		this.addEventHandlers();
+	}
+	

```

Comment:
Nit: There's a lot of trailing whitespace on blank lines in the file. Suggest letting vscode format the file (Alt+Shift+F hotkey).